dsl_parser/rel_graph.py

```diff
@@ -741,8 +742,8 @@ def _node_instance_id(node_id, ctx):
     return new_node_instance_id
 
 
-def _generate_id():
-    return '%05x' % random.randrange(16 ** 5)
+def _generate_id(id_len=5):
+    return ''.join(choice(digits + ascii_letters) for _ in xrange(id_len))
```

Comment:
Thanks for checking this I have no other solid comments.
My comment aimed towards coding style rather then performance since _generate_id() is used sparsely throughout the application life-cycle.